bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_webhook.go

```diff
@@ -115,8 +115,54 @@ func (c *KubeadmConfigSpec) validate(name string) error {
 		knownPaths[file.Path] = struct{}{}
 	}
 
+	allErrs = append(allErrs, c.validateIgnition()...)
+
 	if len(allErrs) == 0 {
 		return nil
 	}
 	return apierrors.NewInvalid(GroupVersion.WithKind("KubeadmConfig").GroupKind(), name, allErrs)
 }
+
+func (c *KubeadmConfigSpec) validateIgnition() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if c.Ignition != nil && c.Format != Ignition {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec", "format"), c.Format, fmt.Sprintf("must be set to %q if spec.ignition is set", Ignition)))
+	}
+
+	if c.Format == Ignition && c.Ignition == nil {
```

Comment:
might be worth to move all the ignition related checks under
 ```
if c.Format != Ignition {
    return allErrs
 }
```

Reply:
Can you please elaborate? I'm not sure what you mean. Two checks we have check if settings are in sync, third condition is a gate for returning early, as there is no point of checking other things at this point. All ignition related checks are already in `validateIgnition()` function. Would you like me to split it further?

Reply:
s/under/below, sorry

The idea hew is that after the if at L139 there is a bunch of ignition specific checks, so I would be nice to move this ignition specific check down as well. 
But this is really cosmetic, feel free to ignore

Reply:
I think existing order of logic is optimal for now, so will mark as resolved.